### PR TITLE
Add Log4Shell recipes to "basic protection set"

### PR DESCRIPTION
**Pull request by seanf**

### CODE REVIEW

File: recipes/Java/Dependencies/Maven/upgrade-log4j-version-dependency.yaml [hunk 1]
```diff
@@ -40,6 +40,24 @@ search:
       matches: 2\.0.*|2\.1(\.\d+)*|2\.[2-9](\.\d+)*|2\.1[0-6]\..*
     tagName:
       is: version
+scopes:
+  library:
+    not:
+      name: {}
```
Comment (wanderneto) [inline recipes/Java/Dependencies/Maven/upgrade-log4j-version-dependency.yaml:46]: Is this line necessary?

File: recipes/Java/Dependencies/Maven/upgrade-log4j-version-property.yaml [hunk 2]
```diff
@@ -26,6 +26,24 @@ search:
       matches: 2\.0.*|2\.1(\.\d+)*|2\.[2-9](\.\d+)*|2\.1[0-6]\..*
     tagName:
       matches: log4j[.-]version|log4j[.-](api|core)[.-]version|org[.-]apache[.-]logging[.-]log4j[.-]version
+scopes:
+  library:
+    not:
+      name: {}
```
Comment (wanderneto) [inline recipes/Java/Dependencies/Maven/upgrade-log4j-version-property.yaml:32]: Is this line necessary?

Comment (wanderneto) [inline recipes/Java/Dependencies/Maven/upgrade-log4j-version-property.yaml:32]: Is this line necessary?

File: recipes/Java/Logging/Log4J/log4j_log4shell.yaml [hunk 3]
```diff
@@ -8,14 +8,27 @@ metadata:
...
   cweCategory: 20
   enabled: true
-  tags: security;SLF4J;OWASP Top 10;framework specific;Log4j;logging;injection
+  tags: Log4j;OWASP Top 10;SLF4J;basic protection set;framework specific;injection;logging;security
 search:
   class: {}
 scopes:
   library:
+    not:
+      name: {}
```
Comment (wanderneto) [inline recipes/Java/Logging/Log4J/log4j_log4shell.yaml:17]: Ditto

**Review by wanderneto — APPROVED** [review]
They seem good to me. Just see if that `name: {}` is necessary or could be deleted.

File: recipes/Java/Dependencies/Maven/upgrade-log4j-version-dependency.yaml (hunk 1 above)
Comment (seanf) [reply recipes/Java/Dependencies/Maven/upgrade-log4j-version-dependency.yaml]: You can't delete from the UI, but I was able to delete it by editing the YAML (without breaking the UI). Amended.

**Comment by seanf** [comment]
Thanks for the review @wanderneto

**Pull request merged**